infrastructure/network/connmanager/connmanager.go

```diff
@@ -119,6 +121,23 @@ func (c *ConnectionManager) connectionsLoop() {
 
 		c.checkIncomingConnections(connSet)
 
+		if len(connSet) == 0 && !cfg.DisableDNSSeed {
```

Comment:
Checking `connSet` (or `connections`) is a bit problematic, since that includes connections that are currently being initiated (and may fail without ever successfully connecting). Also note that DNS-seeding is not concerned with incoming connections at all, and `connSet` does contain those.
It's better to check whether `ConnectionManager.activeOutgoing` is empty instead.